pkg/sql/colexec/evalExpression.go

```diff
@@ -193,16 +199,24 @@ func EvalExpressionOnce(proc *process.Process, planExpr *plan.Expr, batches []*b
 	if err != nil {
 		return nil, err
 	}
+
+	// if memory reuse, we can get it directly because we do only one evaluate.
 	if e, ok := executor.(*FunctionExpressionExecutor); ok {
 		e.resultVector = nil
 		return vec, nil
-	} else {
-		nv, er := vec.Dup(proc.Mp())
-		if er != nil {
-			return nil, er
-		}
-		return nv, nil
 	}
+	if e, ok := executor.(*FixedVectorExpressionExecutor); ok {
+		e.resultVector = nil
+		return vec, nil
+	}
+
+	// I'm not sure if dup is good. but if not.
+	// we should check batch's cnt first, get if it's 1, dup if not.
+	nv, er := vec.Dup(proc.Mp())
```

Comment:
I don't know either, but it definitely should not check cnt.   It is always about ownership, do you transfer or borrow, or hold a shared/read only copy

Reply:
yes. support a more parameter may better.
I will do that in another pr.